### PR TITLE
Feature/modelling mstllgbm

### PR DESCRIPTION
**Pull request by gvalleg**
Incorporate changes from modellingMSTLLGBM into custom_folds

### CODE REVIEW

File: mango_base/mango/dashboards/__init__.py
Comment (antoniogonzalezsuarez) [inline mango_base/mango/dashboards/__init__.py:1]: We will check this in future merge but it seems like should not be here

**Comment by antoniogonzalezsuarez** [comment]
The scripts are fine but I don't know if the repo is the place to be. Please do not remove yet

File: mango_base/mango/dashboards/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-from .file_explorer_app import FileExplorerApp
\ No newline at end of file
+# from .file_explorer_app import FileExplorerApp
```
Comment (gvalleg) [inline mango_base/mango/dashboards/__init__.py:1]: Why?

File: mango_base/mango/dashboards/time_series_app.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    select_agr_tmp,
+                )
+            elif visualization == "Forecast":
+                plot_forecast(forecast, st.session_state["selected_series"])
+                plot_error_visualization(forecast, st.session_state["selected_series"])
+
+
+if __name__ == "__main__":
+    file = "daily_forecast_error.csv"
+    logo_path = r"https://www.multiserviciosaeroportuarios.com/wp-content/uploads/2024/03/cropped-Logo-transparente-blanco-Multiservicios-Aeroportuarios-Maero-1-192x192.png"
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_app.py:87]: This should not be here.

File: mango_base/mango/dashboards/time_series_app.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import pandas as pd
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_app.py:1]: Review if this script .py must be here or in the time-series-dashboard branch.
@antoniogonzalezsuarez

File: mango_base/mango/dashboards/time_series_utils/constants.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Constants for time aggregation
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_utils/constants.py:1]: Review if this script .py must be here or in the time-series-dashboard branch.
@antoniogonzalezsuarez

File: mango_base/mango/dashboards/time_series_utils/data_loader.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import pandas as pd
+import streamlit as st
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_utils/data_loader.py:2]: Review if this script .py must be here or in the time-series-dashboard branch.
@antoniogonzalezsuarez

File: mango_base/mango/dashboards/time_series_utils/data_processing.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import pandas as pd
+from mango_time_series.mango.utils.processing import aggregate_to_input
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_utils/data_processing.py:2]: Review if this script .py must be here or in the time-series-dashboard branch.
@antoniogonzalezsuarez

File: mango_base/mango/dashboards/time_series_utils/file_uploader.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
+import pandas as pd
+import streamlit as st
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_utils/file_uploader.py:2]: Review if this script .py must be here or in the time-series-dashboard branch.
@antoniogonzalezsuarez

File: mango_base/mango/dashboards/time_series_utils/ui_components.py [hunk 8]
```diff
@@ -0,0 +1,378 @@
+import streamlit as st
+import pandas as pd
```
Comment (gvalleg) [inline mango_base/mango/dashboards/time_series_utils/ui_components.py:2]: Review if this script .py must be here or in the time-series-dashboard branch.
@antoniogonzalezsuarez

File: mango_time_series/mango/tests/data/calendar_pivot_keep_com.csv [hunk 9]
```diff
@@ -0,0 +1,1036 @@
... Cataluna,Fiesta Nacional de Espana,Fiesta de Castilla y Leon,Fiesta de la Comunidad de Madrid,Fiesta del Eid Fitr,Fiesta del SacrificioAid Al Adha,Fiesta del SacrificioEidul Adha,Fiesta del Trabajo,Inmaculada Concepcion,Jueves Santo,La Bien Aparecida,Lunes de Fallas,Lunes de Pascua,Lunes siguiente a Ano nuevo,Lunes siguiente a Asuncion de la Virgen,Lunes siguiente a Dia de Andalucia,Lunes siguiente a Dia de Asturias,Lunes siguiente a Dia de Canarias,Lunes siguiente a Dia de Extremadura,Lunes siguiente a Dia de La Rioja,Lunes siguiente a Dia de San Jorge,Lunes siguiente a Dia de la Constitucion Espanola,Lunes siguiente a Dia de la Region de Murcia,Lunes siguiente a Epifania del Senor,Lunes siguiente a Fiesta Nacional de Espana,Lunes siguiente a Fiesta de Castilla y Leon,Lunes siguiente a Fiesta de la Comunidad de Madrid,Lunes siguiente a Fiesta del Trabajo,Lunes siguiente a Inmaculada Concepcion,Lunes siguiente a Natividad del Senor,Lunes siguiente a San Jose,Lunes siguiente a Todos los Santos,Martes de Carnaval,Natividad del Senor,Nuestra Senora de Africa,San Esteban,San Jose,San Juan,Santiago Apostol,Todos los Santos,Traslado de San Jose,V Centenario Vuelta al Mundo,Viernes Santo
```
Comment (gvalleg) [inline mango_time_series/mango/tests/data/calendar_pivot_keep_com.csv:1]: This should not be here.

File: mango_time_series/mango/tests/data/calendar_pivot_keep_com_distances.csv
Comment (gvalleg) [inline mango_time_series/mango/tests/data/calendar_pivot_keep_com_distances.csv:1]: This should not be here.

File: mango_time_series/mango/tests/data/calendar_pivot_keep_com.csv
Comment (gvalleg) [inline mango_time_series/mango/tests/data/calendar_pivot_keep_com.csv:1]: This should not be here.

File: mango_time_series/mango/tests/data/calendar_com_pivot.csv
Comment (gvalleg) [inline mango_time_series/mango/tests/data/calendar_com_pivot.csv:1]: This should not be here.

File: mango_time_series/mango/tests/data/calendar_com.csv
Comment (gvalleg) [inline mango_time_series/mango/tests/data/calendar_com.csv:1]: This should not be here.

File: mango_time_series/mango/tests/data/calendar.csv
Comment (gvalleg) [inline mango_time_series/mango/tests/data/calendar.csv:1]: This should not be here.

File: mango_time_series/mango/utils/processing.py [hunk 10]
```diff
@@ -375,7 +376,7 @@ def process_time_series(
         id_cols=SERIES_CONF["KEY_COLS"],
         freq="d",
         min_max_by_id=True,
-        date_end="2024-09-30",
+        date_end="2024-10-01",
```
Comment (gvalleg) [inline mango_time_series/mango/utils/processing.py:379]: This should be in SERIES_CONF?

File: mango_time_series/script_MSTL.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+from mango_time_series.mango.timeseries_class.base_class import TimeSeriesProcessor
+from mango_time_series.mango.timeseries_class.config_class import SeriesConfiguration
+
+from mango_base.mango.logging.logger import get_basic_logger
+
+logger = get_basic_logger()
+
+# Load the data
+df = pd.read_excel(
+    r"G:\Unidades compartidas\clece_pmr_202207\proyecto\desarrollo\datos\time_series_synthetic_interpolate.xlsx"
```
Comment (gvalleg) [inline mango_time_series/script_MSTL.py:12]: This should not be here.

File: mango_time_series/script_MSTL.py [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+)
+# df["aeropuerto"] = "BCN"
```
Comment (gvalleg) [inline mango_time_series/script_MSTL.py:14]: This should not be here.

File: mango_time_series/script_TBATS.py
Comment (gvalleg) [inline mango_time_series/script_TBATS.py:1]: I think that this file is a project file that we need to remove

File: mango_time_series/script_MSTL.py
Comment (gvalleg) [inline mango_time_series/script_MSTL.py:1]: I think that this file is a project file that we need to remove

File: mango_time_series/script_aLGBM.py
Comment (gvalleg) [inline mango_time_series/script_aLGBM.py:1]: I think that this file is a project file that we need to remove.

File: mango_time_series/script_nLGBM.py
Comment (gvalleg) [inline mango_time_series/script_nLGBM.py:1]: I think that this file is a project file that we need to remove

File: mango_time_series/script_nbeats.py
Comment (gvalleg) [inline mango_time_series/script_nbeats.py:1]: I think that this file is a project file that we need to remove

**Review by gvalleg — COMMENTED** [review]
A lot of changes and files that must not be there.

File: mango_time_series/mango/utils/__init__.py [hunk 13]
```diff
@@ -0,0 +1,2 @@
+from mango_time_series.mango.utils.CONST import timedelta_equivalences, PARAMETERS_BASE
+
```
Comment (ggsdc) [inline mango_time_series/mango/utils/__init__.py:2]: This has to be changed to `from mango_time_series.utils.COSNT import ...`

**Review by ggsdc — CHANGES_REQUESTED** [review]
A lot of changes neede here, and need to update with curent version of develop branch

**Pull request merged**